File: android/app/src/main/java/com/laneshadow/ui/components/atoms/IconSymbol.kt
package com.laneshadow.ui.components.atoms

import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.ArrowBack
import androidx.compose.material.icons.filled.Close
import androidx.compose.material.icons.filled.Favorite
import androidx.compose.material.icons.filled.Home
import androidx.compose.material.icons.filled.Info
import androidx.compose.material.icons.filled.Menu
import androidx.compose.material.icons.filled.Person
import androidx.compose.material.icons.filled.Search
import androidx.compose.material.icons.filled.Settings
import androidx.compose.material.icons.filled.Star
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.platform.testTag
import androidx.compose.ui.semantics.Role
import androidx.compose.ui.semantics.contentDescription
import androidx.compose.ui.semantics.role
import androidx.compose.ui.semantics.semantics
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp

/**
 * Icon name to Material Design icon mapping
 *
 * Maps common MaterialCommunityIcons glyph names to their Material Icons equivalents.
 * Fallback to Icons.Default.Info for unmapped names (no Help icon available).
 */
private fun iconByName(name: String): ImageVector = when (name.lowercase()) {
    "home" -> Icons.Default.Home
    "settings" -> Icons.Default.Settings
    "search" -> Icons.Default.Search
    "heart" -> Icons.Default.Favorite
    "star" -> Icons.Default.Star
    "map" -> Icons.Default.Info // Fallback: no Map icon in default set
    "person" -> Icons.Default.Person
    "close" -> Icons.Default.Close
    "menu" -> Icons.Default.Menu
    "arrow-left" -> Icons.Default.ArrowBack
    else -> Icons.Default.Info // Fallback: no Help icon in default set
}

/**
 * IconSymbol component
 *
 * Cross-platform icon component. On Android, this renders Material Design icons.
 * Following RN wrapper API from react-native/components/ui/icon-symbol.tsx
 *
 * @param name Icon name (MaterialCommunityIcons glyph name)
 * @param size Icon size in dp (default: 24.dp)
 * @param color Icon tint color (required)
 * @param modifier Modifier for the icon
 * @param testID Test ID for UI testing (optional)
 */
@Composable
fun IconSymbol(
    name: String,
    size: Dp = 24.dp,
    color: Color,
    modifier: Modifier = Modifier,
    testID: String? = null,
) {
    val iconVector = iconByName(name)

    val baseModifier = modifier
        .semantics {
            role = Role.Image
            contentDescription = name
        }

    val finalModifier = if (testID != null) {
        baseModifier.testTag(testID)
    } else {
        baseModifier
    }

    Icon(
        imageVector = iconVector,
        contentDescription = name,
        modifier = finalModifier,
        tint = color,
    )
}

File: ios/LaneShadow/Views/Atoms/IconSymbol.swift
import LaneShadowTheme
import SwiftUI

// MARK: - Icon Symbol Component

/**
 * Icon symbol component
 *
 * Cross-platform icon component using SF Symbols with MaterialCommunityIcons name mapping
 * Matrix reference: .spec/prds/native-rewrite/matrices/ui/atoms/IconSymbol.md
 *
 * ## Design Tokens Used
 * - Sizing: Custom size parameter (default: 24)
 * - Colors: Custom color parameter (required)
 * - Typography: `.font(.system(size: size).weight(weight))`
 *
 * ## Icon Name Mapping
 * Maps MaterialCommunityIcons names to SF Symbol equivalents with fallback support
 *
 * ## Parameters
 * - Parameters:
 *   - name: Icon name (MaterialCommunityIcons name, mapped to SF Symbol)
 *   - size: Icon size (default: 24)
 *   - color: Icon tint color (required)
 *   - weight: Icon weight (default: .regular)
 *   - testID: Testing identifier (optional)
 */
public struct LSIconSymbol: View {
    @Environment(\.theme) private var theme

    private let name: String
    private let size: CGFloat
    private let color: Color
    private let weight: Font.Weight
    private let testID: String?

    /// Maps MaterialCommunityIcons names to SF Symbol names
    private static let iconMap: [String: String] = [
        "home": "house",
        "home-outline": "house",
        "cog": "gearshape",
        "cog-outline": "gearshape",
        "magnify": "magnifyingglass",
        "heart": "heart.fill",
        "heart-outline": "heart",
        "star": "star.fill",
        "star-outline": "star",
        "map": "map",
        "account": "person.fill",
        "account-outline": "person",
        "close": "xmark",
        "menu": "line.3.horizontal",
        "arrow-left": "arrow.left",
        "chevron-left": "chevron.left",
        "chevron-right": "chevron.right",
        "chevron-down": "chevron.down",
        "chevron-up": "chevron.up",
        "plus": "plus",
        "plus-circle-outline": "plus.circle",
        "minus": "minus",
        "check": "checkmark",
        "pencil": "pencil",
        "delete": "trash",
        "trash-can-outline": "trash",
        "bell": "bell.fill",
        "bell-outline": "bell",
        "information": "info.circle",
        "wifi-off": "wifi.slash",
        // Map controls icons
        "crosshairs-gps": "crosshair.circle",
        "layers": "square.stack.3d.up",
        "bookmark": "bookmark",
        "bookmark-outline": "bookmark",
        "message-text-outline": "bubble.left.and.bubble.right",
        "map-outline": "map",
        // Weather icons
        "weather-sunny": "sun.max.fill",
        "weather-rainy": "cloud.rain.fill",
        "weather-windy": "wind",
        "weather-cloudy": "cloud.fill",
        "weather-partly-cloudy": "cloud.sun.fill",
        // Map and navigation icons
        "map-marker-distance": "signpost.right",
        "clock-outline": "clock",
        "leaf": "leaf.fill",
    ]

    public init(
        name: String,
        size: CGFloat = 24,
        color: Color,
        weight: Font.Weight = .regular,
        testID: String? = nil
    ) {
        self.name = name
        self.size = size
        self.color = color
        self.weight = weight
        self.testID = testID
    }

    public var body: some View {
        let mappedName = Self.iconMap[name] ?? name

        Image(systemName: mappedName)
            .resizable()
            .scaledToFit()
            .frame(width: size, height: size)
            .foregroundStyle(color)
            .font(.system(size: size).weight(weight))
            .renderingMode(.template)
            .accessibilityLabel(name)
            .accessibilityHidden(false)
    }
}

// MARK: - Preview

#Preview {
    VStack(spacing: 16) {
        LSIconSymbol(name: "home", size: 24, color: .blue)
        LSIconSymbol(name: "heart", size: 32, color: .red, weight: .bold)
        LSIconSymbol(name: "account-outline", size: 48, color: .gray)
        LSIconSymbol(name: "chevron-right", size: 20, color: .green)
        LSIconSymbol(name: "bell", size: 36, color: .orange, weight: .light)
    }
    .padding()
}
